### DISCUSSION TITLE
Vector (Brush) Control

### DISCUSSION THREAD

**Discussion by AQuickSecond** [keiwando/toonsquid#39] [Ideas]
Thank you for filling the gap that Procreate doesn't. It's still a slow workflow compared to industry tools, but you are closing the gap on this quickly.

There are several missing tools that would help expedite the cleanup workflow:

- Line Layer/Color Layer. Having lineart and color on separate layers is a more loose workflow. It would be nice to have the option to automatically create a color layer with invisible, modifiable vector paths based on the original vector layer's path that could be used for quick, non-destructive color work tied to the layer.
<img width="26" alt="ColorlayerPart1" src="https://user-images.githubusercontent.com/113747272/198869530-7ad4f7c7-3e85-4b5d-9ec0-db04d7b5d9d7.PNG">
<img width="260" alt="ColorLayerPart2" src="https://user-images.githubusercontent.com/113747272/198869531-8245f0fe-6cad-4626-8de4-d0c19ba4554b.PNG">

- Set Colors. An option to force color layer colors into a set pallet, with the option to modify color(s) to automatically affect the synced colors. 
<img width="269" alt="Colors" src="https://user-images.githubusercontent.com/113747272/198869525-734c87c5-7bf5-426f-af6d-4ee35e47caa5.PNG">

- Line Art Cleanup. An eraser tool ("Cutter") specifically designed to cut vector paths, with it contextually cleaning lines.
<img width="275" alt="Part1" src="https://user-images.githubusercontent.com/113747272/198869383-d37a5282-250b-429a-a8fc-28c32f410342.PNG">
<img width="232" alt="Part2" src="https://user-images.githubusercontent.com/113747272/198869407-7da46b11-ba66-460f-b026-d27e353fc87d.PNG">
<img width="236" alt="Part3" src="https://user-images.githubusercontent.com/113747272/198869411-22924d5a-fbf2-4351-be53-731c29617bee.PNG">

- Custom gesture mapping. Rather than have options for gestures tied to specific actions, it'd be best to have a list of gestures for the user to choose to tie to their preferred action.
<img width="316" alt="Mapping" src="https://user-images.githubusercontent.com/113747272/198894624-e3ea1375-0504-463e-9756-a7375d606d4e.PNG">

**Comment by keiwando** [comment]
Thank you very much for all of your suggestions!

I'm planning to add a reference layer option for the fill tool with one of the upcoming updates so that you can tag a layer as the "line work" based on which the fill tool can then be used on a different layer. Is this the primary use for those automatically created vector paths as well? (that's ToonBoom Harmony, right?)

**Reply by AQuickSecond** [reply]
That sounds like a great solution to color layer, excited to see it!

Drawn lines in ToonBoom Harmony are vector by default, with those paths being easy to control, simplify, and connect after drawing a path (useful controls that would be nice to have in Toon Squid, but choosing one specific tool, Cut tool saves the most time in my opinion). 
<img width="386" alt="Vector1" src="https://user-images.githubusercontent.com/113747272/199047333-d55192c7-25ed-4a43-8abe-1015d10db5ff.PNG">
<img width="459" alt="Vector2" src="https://user-images.githubusercontent.com/113747272/199047355-3bc4f96a-5430-444f-b88e-a091cbc2f539.PNG">
<img width="416" alt="Vector3" src="https://user-images.githubusercontent.com/113747272/199047370-92c1c085-b040-400e-81af-447d4b4d3cb4.PNG">

In the case of color layers, they do their best to automatically create invisible vector boundaries for filling based on the line art layer, but those boundaries can be adjusted afterwards if they're off the mark, or if additional detailing is wanted on the same layer. Admittedly, dealing with this can be a bit irksome in specific circumstances.
<img width="500" alt="Vector4" src="https://user-images.githubusercontent.com/113747272/199049024-f72ad29d-28d7-4c74-a2bb-0b8beae98e5e.PNG">


For Toon Squid's purposes, just being able to draw on the color layer afterwards should be enough. Though, since the boundaries are linked to defined colors, a set color pallet feature might get awkward if that's ever an interest.

**Comment by ghost** [comment]
A vector art cleanup/cutter tool would make this app even more powerful than it already is. I’m a frame-by-frame animator and I work in Toon Boom Harmony primarily because the art tools are so easy to work with. Being able to cut off overhangs as illustrated above would be incredible and would give me a reason to make ToonSquid on iPad my primary workflow!

**Reply by ota-works** [reply]
I hope the vector point and cutter tool feature is implemented! It would make migrating to iPad so so much easier for us frame-by-frame animators
